File: my-app/src/api/refund.ts
import api from './axios'

function getAuthHeader() {
  const token = localStorage.getItem('access_token')
  return token ? { Authorization: `Bearer ${token}` } : {}
}

export interface RefundResponse {
  id: number
  orderId: number
  studentId: number
  tutorId: number
  amount: number
  disputeId: number | null
  reason: string
  status: 'PENDING_REFUND' | 'TUTOR_PAID' | 'COMPLETED'
  vnpTxnRef: string | null
  vnpTransactionNo: string | null
  vnpResponseCode: string | null
  paymentUrl: string | null
  tutorPaidAt: string | null
  completedAt: string | null
  createdAt: string | null
  updatedAt: string | null
}

export async function getAdminRefunds(status?: string): Promise<RefundResponse[]> {
  const params: Record<string, string> = {}
  if (status) params.status = status
  const res = await api.get('/api/admin/refunds', {
    headers: getAuthHeader(),
    params,
  })
  return res.data
}

export async function createRefund(data: {
  orderId: number
  amount: number
  reason: string
  disputeId?: number | null
}): Promise<RefundResponse> {
  const res = await api.post('/api/admin/refunds', data, {
    headers: getAuthHeader(),
  })
  return res.data
}

export async function createTutorPaymentUrl(refundId: number): Promise<RefundResponse> {
  const res = await api.post(`/api/admin/refunds/${refundId}/payment-url`, null, {
    headers: getAuthHeader(),
  })
  return res.data
}

export async function completeRefund(refundId: number): Promise<RefundResponse> {
  const res = await api.post(`/api/admin/refunds/${refundId}/complete`, null, {
    headers: getAuthHeader(),
  })
  return res.data
}

export async function getStudentRefunds(): Promise<RefundResponse[]> {
  const res = await api.get('/api/student/refunds', {
    headers: getAuthHeader(),
  })
  return res.data
}

export async function getTutorRefunds(): Promise<RefundResponse[]> {
  const res = await api.get('/api/tutor/refunds', {
    headers: getAuthHeader(),
  })
  return res.data
}

export const REFUND_STATUS_LABELS: Record<string, string> = {
  PENDING_REFUND: 'Chờ tutor thanh toán',
  TUTOR_PAID: 'Tutor đã thanh toán',
  COMPLETED: 'Đã hoàn tiền',
}

export const REFUND_REASON_LABELS: Record<string, string> = {
  QUALITY: 'Chất lượng không đảm bảo',
  TUTOR_CANCEL: 'Gia sư hủy lớp',
  STUDENT_CANCEL: 'Học viên hủy',
  NO_SHOW: 'Gia sư không dạy',
  OTHER: 'Khác',
}

File: web/web/src/main/java/com/tmdt/web/entity/Refund.java
package com.tmdt.web.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Table(name = "refunds")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Refund {

    public enum RefundReason {
        QUALITY,            // Chất lượng không đảm bảo
        TUTOR_CANCEL,       // Gia sư hủy lớp
        STUDENT_CANCEL,     // Học viên hủy
        NO_SHOW,            // Gia sư không dạy
        OTHER               // Khác
    }

    public enum RefundStatus {
        PENDING_REFUND,     // Chờ tutor thanh toán tiền hoàn
        TUTOR_PAID,         // Tutor đã thanh toán xong, chờ admin xác nhận
        COMPLETED           // Admin xác nhận hoàn tiền cho học viên xong
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "order_id", nullable = false)
    private Integer orderId;

    @Column(name = "student_id", nullable = false)
    private Integer studentId;

    @Column(name = "tutor_id", nullable = false)
    private Integer tutorId;

    @Column(nullable = false)
    private Double amount;

    @Column(name = "dispute_id")
    private Integer disputeId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RefundReason reason;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private RefundStatus status = RefundStatus.PENDING_REFUND;

    // VNPAY thông tin thanh toán từ tutor
    @Column(name = "vnp_txn_ref")
    private String vnpTxnRef;

    @Column(name = "vnp_transaction_no")
    private String vnpTransactionNo;

    @Column(name = "vnp_response_code")
    private String vnpResponseCode;

    @Column(name = "payment_url", columnDefinition = "TEXT")
    private String paymentUrl;

    @Column(name = "tutor_paid_at")
    private Date tutorPaidAt;

    @Column(name = "completed_at")
    private Date completedAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Date createdAt;

    @Column(name = "updated_at")
    private Date updatedAt;

    @PrePersist
    public void prePersist() {
        Date now = new Date();
        if (status == null) {
            status = RefundStatus.PENDING_REFUND;
        }
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = new Date();
    }
}